dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaYarnApplicationTypeTask.java

```diff
@@ -114,22 +124,37 @@ private void writeYarnAppTypes(
     }
   }
 
-  private List<String> fetchYARNApplicationTypes(ClouderaManagerHandle handle, String clusterName) {
-    String yarnAppTypesUrl =
+  private Set<String> getYarnApplicationTypes(
```

Comment:
I like the idea to encapsulate this logic into this method. I would recommend to rename it because `get*` often used for simple operation, but we have some logic here including http requests. 

`collect*` or `load*` will be more preferable names from my point of view.